From b403a0e8f52a9e13321ebd98a89e6c7ccb135b3a Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Cristian=20One=C8=9B?= <onet.cristian@gmail.com>
Date: Wed, 21 Nov 2012 20:28:22 +0200
Subject: [PATCH 06/22] Use the proper color group depending on the focus state
 of the widget. This results in the same appearence of selected transactions
 as in normal unfocused item views. (cherry picked from commit
 d4040e9b8ea255708e3a0ebecd0259528c88e287)

---
 kmymoney/widgets/transaction.cpp | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/kmymoney/widgets/transaction.cpp b/kmymoney/widgets/transaction.cpp
index 71166f5..ea454df 100644
--- a/kmymoney/widgets/transaction.cpp
+++ b/kmymoney/widgets/transaction.cpp
@@ -227,6 +227,12 @@ bool Transaction::paintRegisterCellSetup(QPainter *painter, QStyleOptionViewItem
     option.state &= ~QStyle::State_HasFocus;
   }
 
+  if (option.widget && option.widget->hasFocus()) {
+    option.palette.setCurrentColorGroup(QPalette::Active);
+  } else {
+    option.palette.setCurrentColorGroup(QPalette::Inactive);
+  }
+
   if (index.column() == 0) {
     option.viewItemPosition = QStyleOptionViewItemV4::Beginning;
   } else if (index.column() == MaxColumns - 1) {
-- 
1.8.1.4

